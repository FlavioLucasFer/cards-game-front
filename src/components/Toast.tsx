import React, { useState } from 'react';
import BootstrapToast, { ToastProps as BootstrapToastProps } from 'react-bootstrap/Toast';
import PropTypes from 'prop-types';

const propTypes = {
    title: PropTypes.string.isRequired,
    message: PropTypes.string,
}

export type ToastProps = BootstrapToastProps & PropTypes.InferProps<typeof propTypes>;

const Toast: React.FC<ToastProps> = props => {
    const {
        title,
        message,
        ...toastProps
    } = props; 

    const [show, setShow] = useState(true);

    const handleOnClose = () => {
        setShow(false);
    }

    return (
        <BootstrapToast 
            show={show}
            onClose={handleOnClose}
            delay={5000}
            autohide
            {...toastProps}
        >
            <BootstrapToast.Header>
                <strong className="me-auto">{title}</strong>
            </BootstrapToast.Header>
            <BootstrapToast.Body className={toastProps.bg === 'dark' ? 'text-white' : ''}>
                {message}
            </BootstrapToast.Body>
        </BootstrapToast>
    );
}

Toast.propTypes = propTypes;

export default Toast;