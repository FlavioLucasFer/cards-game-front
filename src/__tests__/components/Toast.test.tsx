import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Toast from '../../components/Toast';

configure({ adapter: new Adapter() });

describe('Toast tests', () => {
    it('should render properly', () => {
        const wrapper = shallow(
            <Toast 
                title='Test'
            />
        );

        expect(wrapper.debug()).toMatchSnapshot();
    });
});
