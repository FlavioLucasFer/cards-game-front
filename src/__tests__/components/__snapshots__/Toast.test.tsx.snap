// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast tests should render properly 1`] = `
"<Toast show={true} onClose={[Function: handleOnClose]} delay={5000} autohide={true}>
  <ToastHeader closeLabel=\\"Close\\" closeButton={true}>
    <strong className=\\"me-auto\\">
      Test
    </strong>
  </ToastHeader>
  <ToastBody className=\\"\\" />
</Toast>"
`;
